components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesConfig.kt

```diff
@@ -4,14 +4,20 @@
 
 package mozilla.components.feature.top.sites
 
+import mozilla.components.concept.storage.FrecencyThresholdOption
+
 /**
  * Top sites configuration to specify the number of top sites to display and
  * whether or not to include top frecent sites in the top sites feature.
  *
  * @property totalSites A total number of sites that will be displayed.
  * @property includeFrecent If true, includes frecent top site results.
+ * @property frecencyThreshold If [includeFrecent] is true, only visited sites with a frecency
+ * score above the given threshold will be returned. Defaults to 0 to return all visited pages
+ * sorted by most to least frecent.
  */
 data class TopSitesConfig(
     val totalSites: Int,
-    val includeFrecent: Boolean
+    val includeFrecent: Boolean,
```

Comment:
You should simplify this down to a single optional param, something like `frecencyConfig: FrecencyThresholdOption?`. If it's absent, that would be equivalent to `includeFrecent = false`.

Otherwise, you could have an illegal set of values passed into this data class, e.g. `includeFrecent=false, frecencyThreshold=SKIP_ONE_TIME_PAGES`.

This is the pattern we already follow elsewhere (e.g. fxa).

Reply:
Merge the 2 params into `frecencyConfig: FrecencyThresholdOption?`